awx/main/scheduler/task_manager.py

```diff
@@ -511,39 +444,83 @@ def process_pending_tasks(self, pending_tasks):
                 continue
 
             for rampart_group in preferred_instance_groups:
```

Comment:
Even before this line, I think we can assign `controller_node`. I mean, not even in the loop. We don't need to consider the preferred instance groups, that's just over-complicating the problem. All jobs, project updates, container group jobs, regular jobs, all need a control node. If a control node can't be found, you can't dispatch _any_ jobs. You might as well stop processing pending tasks in general.

Reply:
doing this now